lib/chef/formatters/doc.rb

```diff
@@ -336,6 +356,16 @@ def provider_requirement_failed(action, resource, exception, message)
         end
       end
 
+      def deprecation(message, location=caller(2..2)[0])
+        if Chef::Config[:treat_deprecation_warnings_as_errors]
```

Comment:
This could probably do with a comment to explain that `super` will eventually result in the exception-ing.